go/vt/servenv/grpc_server.go

```diff
@@ -325,6 +350,22 @@ func serveGRPC() {
 	})
 }
 
+func getCPUUsage() float64 {
+	percentages, err := cpu.Percent(0, false)
+	if err != nil || len(percentages) == 0 {
+		return 0
+	}
+	return percentages[0]
+}
+
+func getMemoryUsage() float64 {
+	vmStat, err := mem.VirtualMemory()
+	if err != nil {
+		return 0
+	}
+	return vmStat.UsedPercent
```

Comment:
Does this logic all work properly inside containers where there's allocated memory budgets etc? We can't assume all memory is available for the given process here and we need to make sure we honor things like cgroup memory allocations.